### PR TITLE
Forget ball in secondary state modes

### PR DESCRIPTION
**Pull request by jaagut**
## Proposed changes
- Forgets the ball in secondary state modes
- Resets ball filter

## Related issues
bit-bots/bitbots_world_model/pull/19

## Necessary checks
- [x] Run `catkin build`
- [ ] Write documentation
- [ ] Create issues for future work
- [ ] Test on your machine
- [ ] Test on the robot
- [x] Put the PR on our Project board

### CODE REVIEW

File: bitbots_body_behavior/src/bitbots_body_behavior/main.dsd [hunk 1]
```diff
@@ -125,19 +125,19 @@ $ConfigRole
         YES --> $SecondaryStateDecider
             CORNER_KICK --> $SecondaryStateTeamDecider
                 OUR --> $RankToBallNoGoalie + use_time_to_ball:true
-                    FIRST --> @ChangeAction + action:positioning, @LookAtFieldFeatures, @AvoidBallActive, @GoToCornerKickPosition + mode:striker
-                    SECOND --> @ChangeAction + action:positioning, @LookAtFieldFeatures, @AvoidBallActive, @GoToCornerKickPosition + mode:supporter
-                    THIRD --> @ChangeAction + action:positioning, @AvoidBallActive, @GoToDefensePosition
-                OTHER --> @ChangeAction + action:positioning, @LookAtFieldFeatures, @AvoidBallActive, @GoToCornerKickPosition + mode:others
+                    FIRST --> @ChangeAction + action:positioning, @ForgetBall, @LookAtFieldFeatures, @AvoidBallActive, @GoToCornerKickPosition + mode:striker
+                    SECOND --> @ChangeAction + action:positioning, @ForgetBall, @LookAtFieldFeatures, @AvoidBallActive, @GoToCornerKickPosition + mode:supporter
+                    THIRD --> @ChangeAction + action:positioning, @ForgetBall, @LookAtFieldFeatures, @AvoidBallActive, @GoToDefensePosition
```
Comment (jaagut) [inline bitbots_body_behavior/src/bitbots_body_behavior/main.dsd:130]: Is there any reason, `@LookAtFieldFeatures` was not set before?

Comment (SammyRamone) [reply bitbots_body_behavior/src/bitbots_body_behavior/main.dsd:132]: that was wrong, thanks for finding it

**Review by SammyRamone — APPROVED** [review]

File: bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py [hunk 2]
```diff
@@ -281,6 +283,10 @@ def forget_ball(self):
         self.ball_seen_time_teammate = rospy.Time(0)
         self.ball = PointStamped()
         self.ball_teammate = PointStamped()
+        
+        success = self.reset_ball_filter()
+        if not success:
+            rospy.logwarn("Apparently, ball filter could not be reset...",logger_name='bitbots_blackboard')
```
Comment (NFiedler) [inline bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py:289]: With the Trigger, the list is never None. you need to check the success field.

**Review by NFiedler — CHANGES_REQUESTED** [review]

**Comment by jaagut** [comment]
This  needs bit-bots/bitbots_world_model/pull/19

File: bitbots_blackboard/CMakeLists.txt [hunk 3]
```diff
@@ -1,7 +1,19 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.0.2)
 project(bitbots_blackboard)
 
-find_package(catkin REQUIRED COMPONENTS rospy bitbots_docs)
+find_package(catkin REQUIRED COMPONENTS
```
Comment (Flova) [inline bitbots_blackboard/CMakeLists.txt:4]: Why is this added here?

File: bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py [hunk 4]
```diff
@@ -281,6 +283,10 @@ def forget_ball(self):
         self.ball_seen_time_teammate = rospy.Time(0)
         self.ball = PointStamped()
         self.ball_teammate = PointStamped()
+        
+        success, msg = self.reset_ball_filter()
```
Comment (Flova) [inline bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py:287]: We could print the `msg` in this case

File: bitbots_blackboard/CMakeLists.txt (hunk 3 above)
Comment (jaagut) [reply bitbots_blackboard/CMakeLists.txt:4]: As I added `std_srvs` in the ball filter and here, I noticed other things are missing too.

File: bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py (hunk 4 above)
Comment (jaagut) [reply bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py:287]: True

**Review by SammyRamone — APPROVED** [review]

**Review by johagge — APPROVED** [review]
i tested it and it works well

**Review by NFiedler — APPROVED** [review]

**Review by johagge — APPROVED** [review]
i tested it

**Review by Flova — APPROVED** [review]
